### PR TITLE
🐛Re-implemented "Fixed NonXHR GET on amp-form with Async Input Elements"

### PR DESCRIPTION
**Pull request by torch2424**
relates to #20362
relates to #20594

cc/ @leonalicious @aghassemi 

This attempts to re-implement a fix for the original bugfix at: #20362 , but then caused #20594 . This happened because I mistakenly got confused with my own fix, and @cvializ tried to save me, but then I also gave them my mistaken understanding 😢 But here I fix all the test cases, and added example to mimmick the behavior of `https://m.aliexpress.com/shopcart/list.html` which originally found this bug.

Here are some examples of me being extra sure this is all working:

# Before Fix
ampformampbindbeforefix

# After Fix

## Amp Form Example

**Note: The popup blocker gets enabled by amp bind/amp form combo. Don't know if this is expected. And there are some logs which were removed, that simply state if the form submit is called manually**

ampformampbindpopup

## Amp Recaptcha Example

amprecaptchaampbind

## Other Random Verification Screenshots

screen shot 2019-02-04 at 1 18 29 pm
screen shot 2019-02-04 at 1 19 08 pm

### CODE REVIEW

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -527,11 +527,22 @@ export class AmpForm {
           this.urlReplacement_.expandInputValueSync(varSubsFields[i]);
         }
 
-        this.handleNonXhrGet_(event);
+        /**
+         * If the submit was called with an event, then we shouldn't
+         * manually submit the form
+         */
+        let shouldSubmitFormElement = true;
```
Comment (cvializ) [inline extensions/amp-form/0.1/amp-form.js]: nit: You can avoid a branch using the `Boolean` constructor as a function e.g. 

```js
const shouldSubmitFormElement = Boolean(event);
```

File: extensions/amp-form/0.1/test/test-amp-form.js [hunk 2]
```diff
@@ -2322,12 +2327,10 @@ describes.repeated('', {
         expect(ampForm.urlReplacement_.expandInputValueSync)
             .to.have.been.calledWith(canonicalUrlField);
 
-        return whenCalled(form.submit).then(() => {
-          expect(form.submit).to.have.been.calledOnce;
-          expect(clientIdField.value).to.equal('');
-          expect(canonicalUrlField.value).to.equal(
-              'https%3A%2F%2Fexample.com%2Famps.html');
-        });
+        expect(form.submit).to.have.been.called;
```
Comment (cvializ) [inline extensions/amp-form/0.1/test/test-amp-form.js]: Can you help me understand why you replaced `calledOnce` with `called`? We would want to know if the form was submitted multiple times and avoid it.

Comment (torch2424) [reply extensions/amp-form/0.1/test/test-amp-form.js]: Oops, didn't know I did that. Thanks! 😄

File: extensions/amp-form/0.1/amp-form.js (hunk 1 above)
Comment (torch2424) [reply extensions/amp-form/0.1/amp-form.js]: 🔥 💯 Thank you! 😄

File: extensions/amp-form/0.1/amp-form.js [hunk 3]
```diff
@@ -531,10 +531,7 @@ export class AmpForm {
          * If the submit was called with an event, then we shouldn't
          * manually submit the form
          */
-        let shouldSubmitFormElement = true;
-        if (event) {
-          shouldSubmitFormElement = false;
-        }
+        let shouldSubmitFormElement = Boolean(!event);
```
Comment (cvializ) [inline extensions/amp-form/0.1/amp-form.js]: The `!` operator automatically casts to a boolean, so you can omit the constructor if you're negating it.

Comment (torch2424) [reply extensions/amp-form/0.1/amp-form.js]: Awesome, yeah I was thinking I should, but then I think I got pulled into a meeting or something. 😄 Thanks!

**Pull request merged**